include/openssl/core_names.h

```diff
@@ -166,6 +166,9 @@ extern "C" {
 #define OSSL_PKEY_PARAM_FFC_G        "g"
 #define OSSL_PKEY_PARAM_FFC_Q        "q"
 
+/* Elliptic Curve Domain Parameters */
+#define OSSL_PKEY_PARAM_EC_NAME      "name"
```

Comment:
Is that the curve name?  That's fairly unique for EC, I wonder if "curve" might be better (macro `OSSL_PKEY_PARAM_EC_CURVE`)

Reply:
Yes, it's the curve name, I would like `NAME` to stay in the name of the constant, as at some point we might identify the curve by its parameters rather than its conventional name.

Also `EC` stands for Elliptic Curve, so it feels a bit strange to call something "foo_EllipticCurve_CURVE"

Reply:
I revisited this and changed this to

```C
#define OSSL_PKEY_PARAM_EC_NAME      "curve-name"
```

would that be more acceptable?